packages/card/card.tsx

```diff
@@ -0,0 +1,165 @@
+import { forwardRef } from 'react'
+import classNames from 'classnames'
+import Image from 'next/image'
+import Link from 'next/link'
+import ProductBadge from '@hashicorp/react-product-badge'
+import { IconArrowRight24 } from '@hashicorp/flight-icons/svg-react/arrow-right-24'
+import type {
+  CardProps,
+  ThumbnailProps,
+  MetaProps,
+  ContentProps,
+  HeadingProps,
+  ProductBadgesProps,
+  DescriptionProps,
+} from './types'
+import s from './style.module.css'
+
+// ! TODO Add proper typing here
+interface CardCompoundType
+  extends React.ForwardRefExoticComponent<
+    CardProps & React.RefAttributes<HTMLDivElement>
+  > {
+  Thumbnail?: $TSFixMe
+  Meta?: $TSFixMe
+  Content?: $TSFixMe
+  Heading?: $TSFixMe
+  Description?: $TSFixMe
+  ProductBadges?: $TSFixMe
+}
+
```

Comment:
🤔 Adding typing for this card as a compound component has proven tricky.  This is the direction I've seen as a working approach, but obviously without the `$TSFixMe` placeholders 😅 

Just added those to get working builds again and put the approach up for discussion!

Reply:
Update: I've refactored the component to use a different approach.  

Instead of forwarding a `ref` to the parent to use with hover state logic via `forwardRef`, I instead implemented a `hoverHandler` prop and moved the `useHover()` hook into the `Card` component proper. I added a `useEffect()` to `Card` that fires when `isHovered` changes, and executes `hoverHandler` if that's passed into the `Card`.  

This does introduce unnecessary `useEffect()` calls if there's no `hoverHandler` passed to `Card`, but this does sidestep the TS typing issues here (and makes the code a bit more flexible for consumers looking to execute logic on `Card` hover).  I'm looking into how to prevent the unnecessary `useEffect()` calls on no `hoverHandler` passed 🤔